### ISSUE TITLE
[CW-1868] Ability to delete conversations

### ISSUE THREAD

**Issue by iammattmartin** [chatwoot/chatwoot#947]
**Is your feature request related to a problem? Please describe.**
No, not related to a current problem.

**Describe the solution you'd like**
There isn't any ability to delete conversations (or old staff from the master admin login). This may be required in certain circumstances where sensitive information is received and cannot be stored, the message is spam, or due to GDPR, the data must be removed.

**Describe alternatives you've considered**
This may be possible via the database directly but generally such action is unwise, especially if cached elsewhere.

<sub>[CW-1868](https://linear.app/chatwoot/issue/CW-1868/ability-to-delete-conversations)</sub>

**Comment by pranavrajs** [comment]
@iammattmartin Thanks for the suggestion. We will take it up in the current sprint itself. 

To start with we will add a feature to delete the conversation from SuperAdmin console.

**Comment by iammattmartin** [comment]
@pranavrajs Do you know the current slated release for this. We have a few convos with data we'd really not want to keep.

**Comment by pranavrajs** [comment]
@iammattmartin This was removed from v1.7.0 as it was not complete at that time, we will get it done on the v1.9.x release

**Comment by pranavrajs** [comment]
@iammattmartin Allowing the admin/agents to delete a conversation might have an unintended sideeffect. If widget is open and admin tries to delete the conversation, the conversation on the widget becomes orphaned. We need to look at the implication in detail. 

As a starting point we will allow the agent/admin to delete a message which solve the spam messages / GDPR request.

**Comment by pranavrajs** [comment]
The API to delete the message is merged in #1353, the UI would be completed in 1.13.0

**Comment by pranavrajs** [comment]
This will be completed along with the UI changes in Message component.

**Comment by ddlidded** [comment]
Is there any update on this?

**Comment by rihards-simanovics** [comment]
Hi there, I currently looking to implement chatwoot for a semi-gov website where the conversation with users' needs to be deleted after a certain period and the conversation must be reset. I hope this feature will be implemented in the coming future.

**Comment by lafi87** [comment]
It is also very important to have the functionality to clean the conversation, without deleting the contact.

**Comment by jordan26** [comment]
I completely agree this is required, I didn't realise we had no option to delete conversations and now we have 5000 and counting.

**Comment by Micorksen** [comment]
Hey!
This is a pretty cool idea, also, could you make like the merge contact button but for tickets?

**Comment by jacekmichalski** [comment]
+1

**Comment by ssthormess** [comment]
This would be really useful....

**Comment by Wamy-Dev** [comment]
I have so many conversations, and it takes forever to load them...

**Comment by mmojadad** [comment]
It is also very important to have the functionality to clean the conversation, without deleting the contact.
Also i need be able to clean attachments of the conversation by extension file type.

**Comment by hpk42** [comment]
Same here, we are currently experimenting with chatwoot and were surprised there is no way in the UI to get rid of conversations. This is also required from a privacy-perspective as users might request removal of data.  Or is there some method to remove conversations meanwhile?

**Comment by iammattmartin** [comment]
Data removal is not an issue, you can delete the user and all their data goes. You just cannot delete the ONE conversation without losing every past communication. 

It's a 4 year old need. I doubt it'll happen soon like much of the needed core functions of the software.

**Comment by mmojadad** [comment]
@iammattmartin Data removal also is an issue, I need not delete the user, But in user conversations there is more than 80GB data screenshots and movies taken by mobiles and there is no way to delete them by filter file type.
Also another lack of important facilities in Chatwoot , All images upload in .jpg format and Chatwoot not support .webp conversion on fly , In such way each screenshot is around 3MB size . All modern applications support on fly media compression method to reduce size on disk. Also:
 filter conversations by type ( text, image,video )
 per-defined policy for upload file extension type and size limits.

**Comment by Evoltstore** [comment]
It is not mentioned here, but:
- If the Inbox is an email set by IMAP/SMPT, "Delete Conversation" would also delete the same email from the original email folder. 

Using ChatWoot for email management this would be perfect, as spam or unnecessary emails could be deleted from the inbox and you could perform a full email management from the chatwoot platform.

**Comment by matisidler** [comment]
+1 here

**Comment by rihards-simanovics** [comment]
> +1 here

please don't `+1` in comments unless you have anything more to add of use, react with 👍. to the original issue post.

I say this not to be pedantic but to remind that this is not the only project people are watching and +1 generate a tone of spam.

**Comment by hozza** [comment]
I'd like to suggest upgrading the priority of this feature request. :up: 

Currently the selfhosted, and also the paid cloud version, would make it difficult to comply with EU laws (and others) due to the missing delete conversations feature.

e.g. if a customer started a conversion and included sensitive info like credentials, payment info, security vulnerability etc throughout a long convo. It's not possible to delete that conversation. 

I guess each message would need to be deleted individually or some API script? This is not super practical on anything other than a small scale.

**Comment by AlaaYlula** [comment]
I can't find a way to delete the attachment file from chatwoot storage when delete the message!, why we need to keep the file on the storage?

**Comment by laianeramos** [comment]
Any updates? i agree with @mmojadad it's very important to delete the media associated with it, whether in local storage or s3_compatible

**Comment by tempire** [comment]
Chatwoot is such a great product, I'm really surprised at this missing feature. Lack of data removal is a big security problem.

**Comment by HighLiuk** [comment]
Any news on this?

**Comment by ChefAndy** [comment]
hmm... I searched for this because I thought "there's no way you can't delete conversations." Well, throw my hat in the ring for this too i guess.

**Comment by jcmatoskx** [comment]
I would just like for the chance to add that the ability to delete entire conversations is crucial, not just for GDPR compliance and data privacy, but also for efficient inbox management and organization, especially when handling spam or irrelevant emails that start to clutter the inbox. 
Prioritizing this feature would greatly improve workflow, data security, and chatwoot user experience.

@pranavrajs @sojan-official 
Sorry for tagging you guys,
Thanks for listening and for your great work!

**Comment by gjermundnor** [comment]
+1 to this! Needed for GDPR compliance, both manual deletion and automatic deletion. 

Automatic deletion could be choosing number of days conversations should exist before automatically being deleted. 

Bonus if conversations also can be deleted from an API.

**Comment by skyboundTinker** [comment]
I came here to look into this too! Is there any other way, say using SQL queries? 

But please make this a feature from within the admin UI. As poster above suggested. API endpoint for this would be great too! It would be nice to be able to prune the DB from stale and old history. Also GDPR is of concern.

**Comment by Evoltstore** [comment]
Hey @sojan-official  we're really happy with the latest fast development that Chatwoot is seeing, congratulations and also we appreciate you keeping the Milestones updated with what you are working on :) 
We couldn't help but notice that this feature "Delete conversations" however disappeared from the Milestones, it is a pity because it is one that we were looking forward to as we find it quite important in order to finally be able to keep inboxes organized by cleaning up spam messages/emails and also being able to be GDPR compliant. 

Deleting conversations feels like, intuitively, it should be a base feature that one would expect the app should have
Any specific reasons you stopped working on this or any ideas if you guys would pick it back up? 

Thank you very much once again for the attention and advice.

**Comment by Evoltstore** [comment]
@sojan-official  Awesome!! Thank you very much for the fast response.
I do have an important follow-up question related specifically to deleting conversations from mail channels (IMAP/SMTP):

💬 If a conversation originated from an email inbox, (email is setup in chatwoot using IMAP/SMPT) will using the upcoming Delete Conversation function also delete the original email from the source inbox (e.g., Gmail, Outlook, etc.)?
Our use case involves managing emails directly from Chatwoot, and for true email inbox management (spam handling, GDPR requests, etc.), it would be very useful if deleting a conversation in Chatwoot could optionally remove the associated email message from the actual email server (via IMAP DELETE or EXPUNGE).

This would allow:

Full spam cleanup directly from the Chatwoot interface.

More seamless inbox management (without needing to access external email clients).
Better compliance when needing to truly delete unwanted or sensitive content.

❓Is this on the "Delete conversations" roadmap or technically feasible?

There could be an optional toggle per inbox/channel like:

[x] When deleting a conversation, also delete original email from server

We believe this would greatly expand the utility of Chatwoot as an helpdesk system.

Thank  you

**Comment by sojan-official** [comment]
@Evoltstore  We will take this up for upcoming releases.  the current change only focuses on deleting the data from Chatwoot server.

**Comment by Evoltstore** [comment]
> [@Evoltstore](https://github.com/Evoltstore) We will take this up for upcoming releases. the current change only focuses on deleting the data from Chatwoot server.

Thank you for taking this into consideration. I am not sure how this version will work but I remember previously an issue with deleting conversations (by deleting the user as well) that because the channel was connected to the email through SMTP&IMAP the conversation would always be automatically re-created in Chatwoot once again so the only way to delete them fully would be to delete manually from both chatwoot and the email inbox.
